### DISCUSSION TITLE
How can I read the raw data from h265 locally, encapsulate it as rtp.Packet and push it to rtspServer?

### DISCUSSION THREAD

**Discussion by shushi0123** [bluenviron/gortsplib#550] [Questions]
I can obtain complete h265 I /P and B frame data through dsp on the amba platform. How can I encapsulate it into rtp.Packet and finally push it to rtstpServer through the WritePacketRTP method? I sometimes get "short buffer" errors when encapsulating h265 data using the pkt.MarshalTo method, and I sometimes get null pointer errors when I split the data into 1200-size loops.
How do you process local h265 raw data and push it to rtspServer? Is there an example? Thank you.

**Comment by aler9** [comment]
Hello, i added the [client-record-format-h264-from-disk example](https://github.com/bluenviron/gortsplib/blob/main/examples/client-record-format-h264-from-disk/main.go) that shows how to wrap raw H264 frames into RTP packets.

You should be able to use your source and H265 by respectively replacing the MPEG-TS callback with your read logic, and by replacing all H264 appearances with H265.
